pkg/dependencies/factory.go

```diff
@@ -159,7 +159,9 @@ func (f *Factory) WithProvider(clusterConfigFile string, clusterConfig *v1alpha1
 	return f
 }
 
-func (f *Factory) WithProviderFactory(clusterConfig *v1alpha1.Cluster, clusterConfigFile string) *Factory {
+func (f *Factory) WithProviderFactory(clusterConfigFile string) *Factory {
+	clusterConfig, _ := v1alpha1.GetClusterConfig(clusterConfigFile)
```

Comment:
I don't love ignoring this error
Is there something we can do about it?

Reply:
I can return error but for that I'll have to change return values for WithProviderFactory, WithProvider and all the related places. Pl correct me if we can do that without making changes at other places.